src/Microsoft.Identity.Web.TokenAcquisition/ConfidentialClientApplicationBuilderExtension.cs

```diff
@@ -44,6 +45,17 @@ public static async Task<ConfidentialClientApplicationBuilder> WithClientCredent
                 return builder;
             }
 
+            // regardless of credential type being set, bound token requires a certificate
+            if (isTokenBinding)
+            {
+                if (credential.Certificate == null)
```

Comment:
I think this works for a POC, but we probably need more abstraction here. Each credential may, or may not, be compatible with mTLS POP. In particular, I am thinking of the 2 FIC credentials. 

CC @jmprieur @MZOLN